### PR TITLE
fix ocis-1043

### PR DESCRIPTION
**Pull request by fschade**
partially fixes: https://github.com/owncloud/ocis/issues/1043
fixes: https://github.com/owncloud/ocis/issues/884

### CODE REVIEW

**Review by LukasHirt — APPROVED** [review]
I guess cloning the whole `items` instead of one by one wouldn't make a real difference here with performance, right? Since I guess there won't ever be hundreds of nav items :shrug:

**Comment by fschade** [comment]
> I guess cloning the whole `items` instead of one by one wouldn't make a real difference here with performance, right? Since I guess there won't ever be hundreds of nav items 🤷

yes its more costy to do it one by one (but not much it's a low level api). The problem if we do

```
const o = [1, 2, 3]
const n = [...o]
n[0] = 99
```

the result would be [99,2,3]

but if we do 
```
const o = [{foo: 1}]
const n = [...o]
n[0].foo = 99
```

we still mutate the original object in o

objects in arrays get refferenced and not deep copied.

File: src/Phoenix.vue [hunk 1]
```diff
@@ -2,7 +2,17 @@
...
-      <template v-if="!showHeader">
+      <div
+        v-if="user.isAuthenticated && !user.userReady"
+        class="loading-overlay"
+        :style="{
+          backgroundImage: 'url(' + configuration.theme.loginPage.backgroundImg + ')'
+        }"
+        uk-height-viewport
+      >
+        <oc-spinner size="xxxlarge" :aria-label="$gettext('Loading')" class="uk-position-center" />
```
Comment (kulmann) [inline src/Phoenix.vue:13]: From my personal feeling the spinner is too big. I would choose `xlarge` size and border sizes in the css class `10px`.

File: src/store/user.js [hunk 2]
```diff
@@ -142,7 +147,7 @@ const actions = {
     }
     const token = vueAuthInstance.getToken()
     if (token) {
-      init(this._vm.$client, token)
+      await init(this._vm.$client, token)
```
Comment (kulmann) [inline src/store/user.js:150]: The `await` here should not be needed, as it's the last statement within the function.

File: src/store/navigation.js [hunk 3]
```diff
@@ -32,7 +32,9 @@ const mutations = {
   ADD_NAV_ITEM(state, { extension, navItem }) {
     const dynamicNavItems = { ...state.dynamicNavItems }
     dynamicNavItems[extension] = dynamicNavItems[extension] || []
-    const index = dynamicNavItems[extension].findIndex(item => item.name === navItem.name)
+    const index = dynamicNavItems[extension].findIndex(
+      item => item.route.path === navItem.route.path
```
Comment (kulmann) [inline src/store/navigation.js:36]: Thanks for spotting this ❤️

**Review by kulmann — CHANGES_REQUESTED** [review]

**Review by kulmann — APPROVED** [review]
👍

**Pull request merged**